### ISSUE TITLE
Follow aria / accessibility best practices

### ISSUE THREAD

**Issue by axlevxa** [cbopp-art/lightcase#179]
Hello,

I want to suggest an update to lightcase to follow best practices for modal accessibility. This guide to [WCAG 1.0(!)](http://web-accessibility.carnegiemuseums.org/code/dialogs/) says:

>[Best practice is] when the modal is open, a screen reader cannot interact with the modal overlay and the main content behind the modal–only the modal itself.

> Upon clicking the modal-activating button, all the main content should get an `aria-hidden='true'` attribute so that a screen reader will no longer read any of the content. Additionally, the modal should get an `aria-hidden='false'` and go from having `{display: none;}` to `{display: block;}.`

Lightcase toggles the `aria-hidden` attribute depending on if the modal is displayed or not, but as far as I can tell it lacks the final step of having a way to apply toggle on another ID/class (i.e. the main page wrapper). It is very difficult to navigate or use lightcase modals with a keyboard; you can test it on the demo page.

Anyone with advice or who knows more about this, feel free to chime in. It is still used in at least one major wordpress plugin.

**Comment by axlevxa** [comment]
Should have thought a bit more before opening.

Accomplished using hooks; posting for others reference:

```
$('a[data-rel=lightcase]').lightcase({
   onInit : {
      ariaInit: function () {
        document.getElementById("main").setAttribute("aria-hidden", "true");
      }
   },
   onClose : {
      ariaClose: function() {
         document.getElementById("main").setAttribute("aria-hidden", "false");
      }
   }
```

Closing issue.
